pandas/_libs/algos_rank_helper.pxi.in

```diff
@@ -131,45 +131,20 @@ def rank_1d_{{dtype}}(object in_arr, ties_method='average',
     argsorted = _as.astype('i8')
 
     {{if dtype == 'object'}}
-    for i in range(n):
```

Comment:
is this all rolled into the generic rank? (except for the if around the nogil block)?

Reply:
Right.  rank_2d already does it this way, so (besides being less verbose) this makes the patterns consistent.